exec/cnex/lib/textio.c

```diff
@@ -0,0 +1,142 @@
+#include "textio.h"
+#include "enums.h"
+
+#ifdef _MSC_VER
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+#ifdef _WIN32
+#include <io.h>
+#endif
+#include <assert.h>
+#include <iso646.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#include <sys/types.h>
+#endif
+
+#include "cell.h"
+#include "exec.h"
+#include "number.h"
+#include "stack.h"
+#include "nstring.h"
+#include "object.h"
+
+
+static FILE *check_file(TExecutor *exec, void *pf)
+{
+    FILE *f = (FILE *)(pf);
+    if (f == NULL) {
+        exec->rtl_raise(exec, "TextioException.InvalidFile", "", BID_ZERO);
+        // Exception code will return from here, since f is already NULL.
```

Comment:
You don't seem to *actually* check the return value of `check_file` anywhere it is called, to see whether it returns `NULL`.

Reply:
Because it will happen right here, and raise this exception.  Once it does that, the stack will realign itself, and jump back to the caller.  Will it not?  I'm not sure what I'm doing wrong here.

Reply:
This isn't C++, you don't have exceptions. After you call `rtl_raise`, control will return back here, and then what? You return `NULL` from this function.

Reply:
Fixed this in textio and io code, in two separate commits.